controllers/api_v1/keyword_test.go

```diff
@@ -127,6 +127,231 @@ func TestKeywordAPIControllerDbTestSuite(t *testing.T) {
 	suite.Run(t, new(KeywordAPIControllerDbTestSuite))
 }
 
+func (s *KeywordAPIControllerDbTestSuite) TestFetchKeywordWithValidParams() {
+	keyword := models.Keyword{
+		Model:                   &gorm.Model{ID: 1},
+		Keyword:                 "testKeyword",
+		Status:                  models.Pending,
+		LinksCount:              1,
+		NonAdwordsCount:         1,
+		TopPositionAdwordsCount: 1,
+		TotalAdwordsCount:       1,
+		UserID:                  s.user.ID,
+		HtmlCode:                "testHTML",
+		FailedReason:            "",
+	}
+	db.GetDB().Create(&keyword)
+
+	headers := http.Header{}
+	headers.Set("Authorization", "Bearer test-access")
+
+	resp := testHttp.PerformRequest(s.engine, "GET", "/api/v1/keywords/1", headers, nil)
+	respBodyData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Error(err)
+	}
+
+	expectedResp := `{` +
+		`"data":{` +
+		`"id":"1",` +
+		`"type":"keyword",` +
+		`"attributes":{` +
+		`"keyword":"testKeyword",` +
+		`"status":"pending",` +
+		`"links_count":1,` +
+		`"non_adwords_count":1,` +
+		`"non_adword_links":null,` +
+		`"top_position_adwords_count":1,` +
+		`"top_position_adword_links":null,` +
+		`"total_adwords_count":1,` +
+		`"html_code":"testHTML",` +
+		`"failed_reason":""` +
+		`},` +
+		`"relationships":{` +
+		`"user":{` +
+		`"data":{` +
+		fmt.Sprintf(`"id":"%v",`, fmt.Sprint(s.user.ID)) +
+		`"type":"user"` +
+		`}}}}}`
+
+	assert.Equal(s.T(), http.StatusOK, resp.Code)
+	assert.Equal(s.T(), expectedResp, string(respBodyData))
```

Comment:
@gutakk Is it possible to JSON unmarshal this to a map then check it by keys? 💭

Reply:
Fixed in d8d8325cffaed057138c1e699860d0df95883ab0